src/main/java/org/apache/sysds/runtime/data/SparseBlockCOO.java

```diff
@@ -319,8 +319,8 @@ public long size(int rl, int ru, int cl, int cu) {
 		for(int i=rl; i<ru; i++)
 			if( !isEmpty(i) ) {
 				int start = internPosFIndexGTE(i, cl);
-				int end = internPosFIndexGTE(i, cu);
-				nnz += (start!=-1) ? (end-start) : 0;
+				int end = internPosFIndexLTE(i, cu-1);
```

Comment:
Hi @mboehm7,
I just checked this again and it always returned the correct result (i.e., zero). I tested different ranges with no non-zero values for COO, MCSR, MCSC, and CSR.

Reply:
I just don't trust example test cases if there is no good explanation for why this is necessary (and conceptually this change is wrong). I'll try to revert this to get all new test cases working (there must be something else being wrong if they're failing) - which test failed with the old code?

Reply:
In case you mean the test cases from the sparse test package, no one failed. They also worked with the way before I changed it. I just noticed that when testing various matrices that it would not consistently produce correct results. After changing the code the way I did the unit tests also pass.

Reply:
Or is your concern here specifically that I used internPosFIndex ?